.azure-pipelines/common-templates/install-tools-template.yml

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+steps:
+    - task: UseDotNet@2
+      displayName: 'Use .NET Core SDK 2.x'
+      inputs:
+        debugMode: false
+        version: 2.x
+
+    - task: UseDotNet@2
+      displayName: 'Use .NET Core SDK 3.x'
+      inputs:
+        debugMode: false
+        version: 3.x
+
+    - task: UseDotNet@2
+      displayName: 'Use .NET Core SDK 5.x'
+      inputs:
+        debugMode: false
+        version: 5.x
+
+    - task: NuGetToolInstaller@1
+      displayName: 'Install Nuget 5.7'
+      inputs:
+         versionSpec: 5.7.0
+         checkLatest: false # Optional
+
+    - task: NuGetAuthenticate@0
+
+    - task: PowerShell@2
+      displayName: 'Install Powershell Core'
```

Comment:
There is no need to install PowerShell core on an agent. To use PowerShell (core), use `pwsh: true` attribute on the `PowerShell@2` task or the `pwsh` shortcut task. e.g.,
``` yml
- task: PowerShell@2
    inputs:
      targetType: 'inline'
      script: 'Write-Host "I'm running on pwsh core!'
      pwsh: true

# Or

- pwsh: 'Write-Host "I'm running on pwsh core!"'
```